saw-core/src/Verifier/SAW/SharedTerm.hs

```diff
@@ -1403,89 +1586,125 @@ scIntMod sc x y = scGlobalApply sc "Prelude.intMod" [x, y]
 scIntToNat
    :: SharedContext -> Term -> IO Term
 scIntToNat sc x = scGlobalApply sc "Prelude.intToNat" [x]
 
--- primitive natToInt :: Nat -> Integer;
+-- | Create a term computing an @Integer@ from a @Nat@.
+--
+-- > natToInt : Nat -> Integer;
 scNatToInt
    :: SharedContext -> Term -> IO Term
 scNatToInt sc x = scGlobalApply sc "Prelude.natToInt" [x]
 
--- primitive intToBv :: (n::Nat) -> Integer -> bitvector n;
+-- | Create a term computing a bitvector of length n from an @Integer@, if
+-- possible.
+--
+-- > intToBv : (n::Nat) -> Integer -> bitvector n;
 scIntToBv
    :: SharedContext -> Term -> Term -> IO Term
 scIntToBv sc n x = scGlobalApply sc "Prelude.intToBv" [n,x]
 
--- primitive bvToInt :: (n::Nat) -> bitvector n -> Integer;
+-- | Create a term computing an @Integer@ from a bitvector of length n.
+-- This produces the unsigned value of the bitvector.
+--
+-- > bvToInt : (n : Nat) -> bitvector n -> Integer;
 scBvToInt
    :: SharedContext -> Term -> Term -> IO Term
 scBvToInt sc n x = scGlobalApply sc "Prelude.bvToInt" [n,x]
 
--- primitive sbvToInt :: (n::Nat) -> bitvector n -> Integer;
+-- | Create a term computing an @Integer@ from a bitvector of length n.
+-- This produces the 2's complement signed value of the bitvector.
+--
+-- > sbvToInt : (n : Nat) -> bitvector n -> Integer;
 scSbvToInt
    :: SharedContext -> Term -> Term -> IO Term
 scSbvToInt sc n x = scGlobalApply sc "Prelude.sbvToInt" [n,x]
 
 
 -- Primitive operations on bitvectors
 
--- | bitvector :: (n : Nat) -> sort 1
--- bitvector n = Vec n Bool
+-- | Create a term computing the type of a length-n bitvector.
+--
+-- > bitvector : (n : Nat) -> sort 1
 scBitvector :: SharedContext -> Natural -> IO Term
 scBitvector sc size = do
   c <- scGlobalDef sc "Prelude.bitvector"
   s <- scNat sc size
   scApply sc c s
 
--- | bvNat :: (x :: Nat) -> Nat -> bitvector x;
+-- | Create a term computing a bitvector of length x from a @Nat@, if possible.
+--
+-- > bvNat : (x : Nat) -> Nat -> bitvector x;
 scBvNat :: SharedContext -> Term -> Term -> IO Term
 scBvNat sc x y = scGlobalApply sc "Prelude.bvNat" [x, y]
 
--- bvToNat :: (n :: Nat) -> bitvector n -> Nat;
+-- | Create a term computing a @Nat@ from a bitvector of length n.
+--
+-- > bvToNat : (n : Nat) -> bitvector n -> Nat;
 scBvToNat :: SharedContext -> Natural -> Term -> IO Term
 scBvToNat sc n x = do
     n' <- scNat sc n
     scGlobalApply sc "Prelude.bvToNat" [n',x]
 
--- | Returns constant bitvector.
+-- | Create a term computing a bitvector of the given length representing the
+-- given 'Integer' value (if possible).
 scBvConst :: SharedContext -> Natural -> Integer -> IO Term
 scBvConst sc w v = assert (w <= fromIntegral (maxBound :: Int)) $ do
   x <- scNat sc w
   y <- scNat sc $ fromInteger $ v .&. (1 `shiftL` fromIntegral w - 1)
   scGlobalApply sc "Prelude.bvNat" [x, y]
 
+-- TODO: This doesn't appear to be used anywhere, and "FinVal" doesn't appear
+-- in Prelude.sawcore... can this be deleted?
```

Comment:
If there is no `FinVal` in `Prelude.sawcore`, then you're right: We should get rid of it. Perhaps that's out of scope for this PR, though?